catalog/security/catalog-security-policyplugin/src/main/java/org/codice/ddf/catalog/security/CatalogPolicy.java

```diff
@@ -0,0 +1,274 @@
+    }
+
+    /**
+     * Get the KeyValuePermission that have been parsed from the permissions string
+     *
+     * @return
+     */
+    public Map<String, Set<String>> getReadPermissionMap() {
+        return Collections.unmodifiableMap(readPermissionMap);
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setCreatePermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.createPermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, createPermissionMap);
+        }
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setUpdatePermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.updatePermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, updatePermissionMap);
+        }
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setDeletePermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.deletePermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, deletePermissionMap);
+        }
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setReadPermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.readPermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, readPermissionMap);
+        }
+    }
+
+    /**
+     * Parses a string  array representation of permission attributes
+     *
+     * @param permStrings String array of permissions to parse
+     */
+    private void parsePermissionsFromString(String[] permStrings, Map<String, Set<String>> permissions) {
+        permissions.clear();
+        if (permStrings != null) {
+            for (String perm : permStrings) {
+                String[] parts = perm.split("=");
+                if (parts.length == 2) {
+                    String attributeName = parts[0];
+                    String attributeValue = parts[1];
+                    String[] attributeValues = attributeValue.split(",");
+                    permissions.put(attributeName,
+                            new HashSet<>(Arrays.asList(attributeValues)));
+                }
+            }
+        }
+    }
+
+    @Override
+    public PolicyResponse processPreCreate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        if (Requests.isLocal(properties)) {
+            return new PolicyResponseImpl(createPermissionMap, null);
```

Comment:
Seems like all of these instances of `PolicyResponseImpl` should be created with unmodifiable copies of these maps instead of leaking them out.